### PR TITLE
Fixed erroneous example.

### PR DESCRIPTION
**Pull request by jfjlaros**
## What This PR Changes
This PR addresses two issues with an example in the tutorial style guide:
- `TRUE` is not defined.
- Redundant comparisons in conditions are [not recommended](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es87-dont-add-redundant--or--to-conditions).

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

File: content/learn/08.contributions/00.arduino-writing-style-guide/arduino-writing-style-guide.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ if (somethingIsTrue) doSomething;
 Instead, use this:
 
 ```
-if (somethingIsTrue == TRUE) {
+if (somethingIsTrue) {
```
Comment (karlsoderby) [inline content/learn/08.contributions/00.arduino-writing-style-guide/arduino-writing-style-guide.md:62]:
```suggestion
if (somethingIsTrue == true) {
```

Comment (karlsoderby) [reply content/learn/08.contributions/00.arduino-writing-style-guide/arduino-writing-style-guide.md:62]: Hey @jfjlaros , I suggest that we keep the `== true` in this if statement, as for less experienced coders, it describes what is happening a bit better.

While `if(somethingIsTrue){}` works the same and involves less code, it is harder to understand for a beginner, as the state it is compared (true) is not clear. If you want to, we can add this as a paragraph below.

Thanks a lot for your submission, let me know what you think.

Comment (jfjlaros) [reply content/learn/08.contributions/00.arduino-writing-style-guide/arduino-writing-style-guide.md:62]: I understand your concerns, but I am of the opinion that we should not encourage this practice. As mentioned in the C++ core guidelines, it is redundant and opens opportunities for errors. The current issue is a good example of this.

I also think that adding redundant truth tests will get the wrong message across for beginners, as they may get the impression that `if` is some sort of function that takes two operands.

As a compromise, maybe the example could be changed to one that uses a meaningful comparison, e.g.,

```cpp
if (distance > 10) {
  // Go nearer.
}
```

Then we could leave the choice of using `== true` up to the author of the tutorial.

**Review by karlsoderby — APPROVED** [review]
Thank you @jfjlaros, I think this looks good.

**Pull request merged**